### ISSUE TITLE
fix: `kaitai_struct_compiler` upstream issues

### ISSUE THREAD

**Issue by ManasviGoyal** [det-lab/kaitai_struct_awkward_runtime#15]
After merging `kaitai_struct_compiler` upstream, the `kaitai-struct-compiler` tool fails. With the upstream, we also need to reflect the new changes in CppCompiler.scala to AwkwardCompiler.scala since the later is a modified copy of the former.

**Comment by zonca** [comment]
> After merging `kaitai_struct_compiler` upstream

what 2 branches are you merging?

> the `kaitai-struct-compiler` tool fails. 

what is the error? fails compiling which file?

> With the upstream, we also need to reflect the new changes in CppCompiler.scala to AwkwardCompiler.scala since the later is a modified copy of the former.

it looks like we need to port the new commits affecting `CppCompiler.scala` to `AwkwardCompiler.scala`

**Comment by ManasviGoyal** [comment]
> it looks like we need to port the new commits affecting `CppCompiler.scala` to `AwkwardCompiler.scala`

@zonca Yes exactly. The current branch `ManasviGoyal/kaitai_awkward_target` is not synced with the latest changes in `kaitai-struct-compiler`. Those changes need to be added.

When I tried making these changes, the generated `.jar` files were not able to parse the `.ksy` files to generate the source and header files.

**Comment by zonca** [comment]
updated compiler to 0.11 and rebased all changes on top of that.
https://github.com/det-lab/kaitai_struct_compiler/pull/9
